internals/idkey/idkey.go

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// idkey supplies an identity key for a machine, container or device. This
+// package provides an implementation based on an Ed25519 based key, currently
+// only supporting a file based key storage solution. This can later be
+// extended to support more secure hardware backed keystores, such as TPM,
+// OP-TEE or YubiKey.
+package idkey
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rand"
+	"crypto/sha512"
+	"crypto/x509"
+	"encoding/base32"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+const identityKeyFile = "key.pem"
+
+// Identity key must implement a crypto signer.
+var _ crypto.Signer = (*IDKey)(nil)
+
+type IDKey struct {
+	keyDir string
+	key    any
```

Comment:
Why is this `any` and not `crypto.Signer`, which is what several methods need anyway. See also comment in `load`.

Reply:
Yes. In fact, I think just making this ```ed25519.PrivateKey``` is even more precise ?

Reply:
Ben indicated he is happy. Resolving.